### PR TITLE
Signup: add ability to test new Page Builder experience

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request

Add support for the upcoming Page Builder abtest

**Note**: This is to allow us to test this more broadly internally. It will currently only work in dev and staging environments. The functions in `lib/signup/page-builder` are annotated with how the `abtest` functionality will work once we get to that point.

#### Testing instructions

* Apply D25718-code to your sandbox
* Sandbox `public-api.wordpress.com` and the site you are going to create in signup (eg `mycooltestingsite.wordpress.com`
* Go to `http://calypso.localhost:3000/start/onboarding` and choose Business in the first step
* Complete signup
* Verify that you land on `/block-editor/page/YOURSITE/2` and that you see a functional Site Header block a the top of the editor like this:
  image
* Try signing up with the other flows and ensure that the Page Builder experience is _not_ loaded.

### CODE REVIEW

File: client/lib/signup/page-builder.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export function shouldEnterPageBuilder() {
+	// This will become an abtest when we are ready to launch
+	inTest = config.isEnabled( 'signup/page-builder' );
+	// using `inTest` this way ensures that `isInPageBuilderTest` will only
+	// return true after this, like assigning the test and later checking `getABTestVariation`
+	return inTest;
+}
+
+export function isElibigleForPageBuilder( segment, flowName ) {
```
Comment (ramonjd) [inline client/lib/signup/page-builder.js]: _is**Eligible**ForPageBuilder_ ?

**Comment by ramonjd** [comment]
I doubt that I've tested this thoroughly enough, because I can't find anything to say other than good job! 🙌 So, `> good job!`

✅ Created a business site: the page editor loaded as expected. I edited the page and block attributes in both desktop and mobile viewport widths. 

✅ Created a non-business site and the page editor was **not** loaded, as expected.

❓ Switched my locale to `it` and followed the instructions above. Logging shows that `newSiteParams.options.in_page_builder` [was not set](https://github.com/Automattic/wp-calypso/pull/31822/files#diff-160d380c201b4db2d947c495e877a98aR194), as expected, but the  page builder loaded in the editor anyway.

File: client/lib/signup/page-builder.js (hunk 1 above)
Comment (mattwiebe) [reply client/lib/signup/page-builder.js]: `eligibly` :shipit: 

The pros and cons of copy-pasta!

**Comment by mattwiebe** [comment]
Thanks for the look @ramonjd 🥇 

> Switched my locale to it and followed the instructions above. Logging shows that newSiteParams.options.in_page_builder was not set, as expected, but the page builder loaded in the editor anyway

I can't replicate this, which is good I guess?

**Comment by mattwiebe** [comment]
(Noting that the failing desktop build is unrelated)

**Comment by ramonjd** [comment]
> I can't replicate this, which is good I guess?

> I tried to reproduce @ramonjd's issue (with es), but was only ever shown the page title (no site builder)

I feel special 😇 Thanks for double-checking. Wonder if it's account-related. It's an older test account. Anyway, carry on!

**Pull request merged**